src/components/Workspace.jsx

```diff
@@ -0,0 +1,258 @@
+import * as lucyEyes from '../assets/animations/lucyEyes.json';
+import * as lucyIdle from '../assets/animations/lucyIdle.json';
+import * as lucyJump from '../assets/animations/lucyJump.json';
+import * as lucySmall from '../assets/animations/lucySmall.json';
+
+
+
+
+import Util from '../assets/util';
+
+class Workspace extends Component {
+    constructor() {
+        super();
+        this.state = {
+            shouldRenderSound: false,
+            songArray: [
+                freddieA,
+                geoffB,
+                lucyC,
+                freddieD,
+                lucyE,
+                geoffD,
+                geoffC,
+                lucyA,
+                freddieB,
+            ],
+            currentSong: null,
+            index: 0,
+            isStopped: false,
+            isPaused: false,
+            animationsArray: null,
+            currentLucyAnimation: lucyIdle,
+            currentFreddieAnimation: freddieIdle,
+            currentGeoffAnimation: geoffIdle,
+            animationsList: [
+                lucyBows,
+                lucyBowSpin,
+                lucyEyes,
+                lucyJump,
+                lucySmall,
+                lucyIdle,
+                geoffGlasses,
+                geoffJump,
+                geoffSpin,
+                geoffBig,
+                geoffCrown,
+                geoffIdle,
+                freddieDrip,
+                freddieEyes,
+                freddieSmall,
+                freddieSpin,
+                freddieThumbs,
+                freddieIdle,
+            ]
+        };
+    }
+
+    getArrayAndHandleSound() {
+        let animations = Util.createAnimationsArray(this.state.songArray);
+        this.setState({
+            animationsArray: animations,
+            index: 0,
+        }, () => {
+            this.handlePlaySound();
+        });
+    }
+
+    handlePlaySound() {
+        this.setState({
+            shouldRenderSound: true,
+            currentSong: this.state.songArray[this.state.index],
+            index: this.state.index + 1,
+        }, () => {
+            console.log(this.state)
+        });
+    }
+
+    handleSongFinishedPlaying = () => {
+        if (this.state.index <= this.state.songArray.length - 1) {
+            this.setState({
+                currentSong: this.state.songArray[this.state.index],
+                index: this.state.index + 1,
+                currentLucyAnimation: lucyIdle,
+                currentGeoffAnimation: geoffIdle,
+                currentFreddieAnimation: freddieIdle,
+            })
+        } else {
+            this.setState({
+                currentSong: null,
+                currentLucyAnimation: lucyIdle,
+                currentGeoffAnimation: geoffIdle,
+                currentFreddieAnimation: freddieIdle,
+            })
+        }
+    }
+
+    handleAnimation = () => {
+        if (this.state.index <= this.state.animationsArray.length) {
```

Comment:
you have `this.state.animationsArray.length - 1` on line 121 are these both right

Reply:
yeah index gets increased before the other one is called so it doesnt work otherwise. dont want to change where index gets increased